### ISSUE TITLE
val_loss nan issue when using val_sample_weights in model.fit

### ISSUE THREAD

**Issue by nolsigan** [keras-team/keras#7917]
When using val_sample_weights in model.fit,
if one of the row in val_sample_weights has all zeros, it produces nan val_loss.

I think it will be nice to show Warning to users who experience this issue,
cause it took me a long time to find what caused nan val loss.
